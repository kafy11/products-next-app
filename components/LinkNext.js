import Link from 'next/link';

const LinkNext = ({ children, href, ...rest }) => (
  <Link href={href}>
    <a {...rest}>
      {children}
    </a>
  </Link>
);

export default LinkNext;