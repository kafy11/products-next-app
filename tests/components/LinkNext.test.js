import { shallow } from 'enzyme';
import LinkNext from '../../components/LinkNext';

let wrapper;
const props = {
    children: "Clique",
    href: "localhost:3000",
    testProp: 9,
};

beforeEach(() => {
    wrapper = shallow(<LinkNext {...props} />);
});

describe('LinkNext', () => {
    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    it('should render the children correctly', () => {
        const children = wrapper.find('a').prop('children');
        expect(children).toBe(props.children);
    });
    
    it('should render the href correctly', () => {
        const href = wrapper.find('Link').prop('href');
        expect(href).toBe(props.href);
    });
    
    it('should add the extra props to the "a" tag', () => {
        const testProp = wrapper.find('a').prop('testProp');
        expect(testProp).toBe(props.testProp);
    });
});