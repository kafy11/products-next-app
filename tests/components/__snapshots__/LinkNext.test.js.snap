// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkNext should render correctly 1`] = `
<Link
  href="localhost:3000"
>
  <a
    testProp={9}
  >
    Clique
  </a>
</Link>
`;
